File: PTT-GC-API/Models/Owner/Owner.cs
using NPOI.Util;
using System.ComponentModel.DataAnnotations;
using System.Numerics;
using BigInteger = System.Numerics.BigInteger;

namespace PTT_GC_API.Models.Owner
{
    public class Owner
    {
        public int Id { get; set; }
        [StringLength(255)]
        public string OwnerName { get; set; }
        [StringLength(255)]
        public string EmployeeID { get; set; }
        [StringLength(255)]
        public string FirstName { get; set; }
        [StringLength(255)]
        public string LastName { get; set; }
        [StringLength(255)]
        public string Indicator { get; set; }
        [StringLength(50)]
        public string Telephone { get; set; }
        [StringLength(255)]
        public string Email { get; set; }

        //2020-06-16 For Org. Chart
        [StringLength(255)]
        public string Title { get; set; }
        [StringLength(10)]
        public string BloodGrp { get; set; }
        [StringLength(255)]
        public string Extension { get; set; }
        public int? CompanyCode { get; set; }

        [StringLength(50)]
        public string CompanyShortTxt { get; set; }
        [StringLength(255)]
        public string CompanyName { get; set; }
        [StringLength(255)]
        public string EmpGroup { get; set; }
        [StringLength(255)]
        public string EmpGroupTxt { get; set; }
        public int? EmpSubGroup { get; set; }

        [StringLength(255)]
        public string EmpSubGroupTxt { get; set; }
        public int? EmploymentStatus { get; set; }
        [StringLength(255)]
        public string EmploymentStatusTxt { get; set; }
        [StringLength(255)]
        public string AdminGroup { get; set; }
        public long? MainPositionCostCenter { get; set; }
        public int? AssignmentCostCenter { get; set; }
        [StringLength(255)]
        public string ActionType { get; set; }
        [StringLength(255)]
        public string ActionText { get; set; }
        public int? OrgID { get; set; }
        [StringLength(255)]
        public string OrgTextEN { get; set; }
        [StringLength(50)]
        public string OrgShortTextEN { get; set; }
        public int? OrgLevel { get; set; }
        public int? PositionID { get; set; }
        [StringLength(255)]
        public string PositionTextEN { get; set; }
        [StringLength(50)]
        public string PositionShortTextEN { get; set; }
        public int? PositionLevel { get; set; }
        public bool? ManagerialFlag { get; set; }
        public bool? MainPositionFlg { get; set; }
        public int? ParentOrgID { get; set; }
        public int? UnitOrgID { get; set; }
        [StringLength(50)]
        public string UnitShortTextEN { get; set; }
        [StringLength(255)]
        public string UnitTextEN { get; set; }
        public int? UnitManagerPositionID { get; set; }
        public int? UnitManagerEmpID { get; set; }
        public int? SupOrgID { get; set; }
        [StringLength(255)]
        public string SupShortTextEN { get; set; }
        [StringLength(255)]
        public string SupTextEN { get; set; }
        public int? SupManagerPositionID { get; set; }
        public int? SupManagerEmpID { get; set; }
        public int? ShiftOrgID { get; set; }
        [StringLength(50)]
        public string ShiftShortTextEN { get; set; }
        [StringLength(255)]
        public string ShiftTextEN { get; set; }
        public int? ShiftManagerPositionID { get; set; }
        public int? ShiftManagerEmpID { get; set; }
        public int? DivOrgID { get; set; }
        [StringLength(255)]
        public string DivShortTextEN { get; set; }
        [StringLength(255)]
        public string DivTextEN { get; set; }
        public int? DivManagerPositionID { get; set; }
        public int? DivManagerEmpID { get; set; }
        public int? DepOrgID { get; set; }
        [StringLength(255)]
        public string DepShortTextEN { get; set; }
        [StringLength(255)]
        public string DepTextEN { get; set; }
        public int? DepManagerPositionID { get; set; }
        public int? DepManagerEmpID { get; set; }
        public int? FNOrgID { get; set; }
        [StringLength(50)]
        public string FNShortTextEN { get; set; }
        [StringLength(255)]
        public string FNTextEN { get; set; }
        public int? FNManagerPositionID { get; set; }
        public int? FNManagerEmpID { get; set; }
        public int? FNGRPOrgID { get; set; }
        [StringLength(50)]
        public string FNGRPShortTextEN { get; set; }
        [StringLength(255)]
        public string FNGRPTextEN { get; set; }
        public int? FNGRPManagerPositionID { get; set; }
        public int? FNGRPManagerEmpID { get; set; }
        public int? PSDOrgID { get; set; }
        [StringLength(50)]
        public string PSDShortTextEN { get; set; }
        [StringLength(255)]
        public string PSDTextEN { get; set; }
        public int? PSDManagerPositionID { get; set; }
        public int? PSDManagerEmpID { get; set; }
        public int? CEOOrgID { get; set; }
        [StringLength(50)]
        public string CEOShortTextEN { get; set; }
        [StringLength(255)]
        public string CEOTextEN { get; set; }
        public int? CEOManagerPositionID { get; set; }
        public int? CEOManagerEmpID { get; set; }
        [StringLength(255)]
        public string DataSource { get; set; }
        [StringLength(255)]
        public string Workstream { get; set; }
    }
}

File: PTT-GC-SPA/src/app/core/models/owner.ts
// import { DropdownBu } from './owner';
export interface Owner {
  ownerName: string;
  email: string;
  telephone: string;
}


export class OwnerModel {
  constructor() { }

  public id: number;
  public ownerName: string;
  public employeeID: string;
  public firstName: string;
  public lastName: string;
  public indicator: string;
  public telephone: string;
  public email: string;
  public title: string;
  public bloodGrp: string;
  public extension: string;
  public companyCode: number;
  public companyShortTxt: string;
  public companyName: string;
  public empGroup: string;
  public empGroupTxt: string;
  public empSubGroup: number;
  public empSubGroupTxt: string;
  public employmentStatus: number;
  public employmentStatusTxt: string;
  public adminGroup: string;
  public mainPositionCostCenter: number;
  public assignmentCostCenter: number;
  public actionType: string;
  public actionText: string;
  public orgID: number;
  public orgTextEN: string;
  public orgShortTextEN: string;
  public orgLevel: number;
  public positionID: number;
  public positionTextEN: string;
  public positionShortTextEN: string;
  public positionLevel: number;
  public managerialFlag: boolean;
  public mainPositionFlg: boolean;
  public parentOrgID: number;
  public unitOrgID: number;
  public unitShortTextEN: string;
  public unitTextEN: string;
  public unitManagerPositionID: number;
  public unitManagerEmpID: number;
  public supOrgID: number;
  public supShortTextEN: string;
  public supTextEN: string;
  public supManagerPositionID: number;
  public supManagerEmpID: number;
  public shiftOrgID: number;
  public shiftShortTextEN: string;
  public shiftTextEN: string;
  public shiftManagerPositionID: number;
  public shiftManagerEmpID: number;
  public divOrgID: number;
  public divShortTextEN: string;
  public divTextEN: string;
  public divManagerPositionID: number;
  public divManagerEmpID: number;
  public depOrgID: number;
  public depShortTextEN: string;
  public depTextEN: string;
  public depManagerPositionID: number;
  public depManagerEmpID: number;
  public fNOrgID: number;
  public fNShortTextEN: string;
  public fNTextEN: string;
  public fNManagerPositionID: number;
  public fNManagerEmpID: number;
  public fNGRPOrgID: number;
  public fNGRPShortTextEN: string;
  public fNGRPTextEN: string;
  public fNGRPManagerPositionID: number;
  public fNGRPManagerEmpID: number;
  public pSDOrgID: number;
  public pSDShortTextEN: string;
  public pSDTextEN: string;
  public pSDManagerPositionID: number;
  public pSDManagerEmpID: number;
  public cEOOrgID: number;
  public cEOShortTextEN: string;
  public cEOTextEN: string;
  public cEOManagerPositionID: number;
  public cEOManagerEmpID: number;
  public dataSource: string;
}

export interface IOwner {
  userManagementID: string;
  employeeID: string;
  ownerName: string;
  firstName: string;
  lastName: string;
  email: string;
  positionID: string;
  positionLevel: string;
  positionShortTextEN: string;
  positionTextEN: string;
  buID: string;
  buText: string;
  workstreamID: string;
  workstreamTitle: string;
  roleId: string;
  roleName: string;
  createOn: string;
  remark: string;
  active: boolean;
}

export interface IOwnerInitiative {
  employeeID: string;
  firstName: string;
  lastName: string;
  positionID: string;
  email: string;
  buid: string;
  workstreamID: string;
  roleID: string[];
  remark: string;
  dropdownBu: DropdownBu[];
  dropdownPosition: DropdownPosition[];
  dropdownWorkstream: DropdownWorkstream[];
}

export interface DropdownBu {
  value: string;
  text: string;
}

export interface DropdownPosition {
  value: string;
  text: string;
}

export interface DropdownWorkstream {
  value: string;
  text: string;
}

export interface OwnerList {
  ownerName: string;
  employeeID: string;
  firstName: string;
  lastName: string;
  indicator: string;
  telephone: string;
  email: string;
}